src/proc/Vigad.ts

```diff
@@ -52,43 +53,136 @@ export class Vigad {
     }
 
     /**
-     * Add a new capture area. Returns the id of the capture area.
-     * @param width
-     * @param height
-     * @param top
-     * @param left
-     * @return index: number
+     * Add a new capture area with the specified dimensions and position.
+     * @param {number} width - The width of the capture area.
+     * @param {number} height - The height of the capture area.
+     * @param {number} top - The top position of the capture area.
+     * @param {number} left - The left position of the capture area.
+     * @returns {number} - The ID of the newly added capture area.
      */
     public addCaptureArea(
-        width: number,
-        height: number,
-        top: number,
-        left: number
-    ): number {
+        width = 100,
+        height = 100,
+        top = 0,
+        left = 0
+    ): string {
         const ca = new CaptureArea(width, height, top, left);
+
+        ca.setId(this.generateCaptureAreaId());
         this.captureAreas.push(ca);
-        ca.setId(this.captureAreas.length - 1);
         this.tesseractHandler.enableCaptureArea(ca);
+
         return ca.getId();
     }
 
     /**
-     * Delete a capture area by id
-     * @param id
-     * @return void
+     * Generates a unique capture area ID.
+     * @returns {string} - The generated capture area ID.
      */
-    public deleteCaptureArea(id: number): void {
-        this.captureAreas.splice(id, 1);
-        this.tesseractHandler.removeWorker();
+    public generateCaptureAreaId(): string {
+        const {
+            numbers,
+            lowercaseLetters,
+            uppercaseLetters,
+            generateValidToken,
+        } = useTokenGenerator();
+
+        let caId: string;
+
+        do {
+            caId = generateValidToken(
+                6,
+                numbers + lowercaseLetters + uppercaseLetters,
+                true,
+                true,
+                false,
+                true,
+                false
+            );
+        } while (this.captureAreaIdExists(caId));
+
+        return caId;
     }
 
     /**
-     * Get a capture area by id
-     * @param id
-     * @return CaptureArea
+     * Check if the capture area ID already exists.
+     * @param {string} caId - The capture area ID to check.
+     * @returns {boolean} - Returns true if the capture area ID exists, false otherwise.
      */
-    public getCaptureArea(id: number): CaptureArea {
-        return this.captureAreas[id];
+    private captureAreaIdExists(caId: string): boolean {
+        return this.captureAreas.some((area) => area.getId() === caId);
+    }
+
+    /**
+     * Renames a capture area by updating its ID.
+     * @param {string} oldId - The old ID of the capture area.
+     * @param {string} newId - The new ID to assign to the capture area.
+     * @returns {boolean} - Returns true if the renaming was successful, false otherwise.
+     */
+    public renameCaptureArea(oldId: string, newId: string): boolean {
```

Comment:
Some weird stuff is happening here.
Suppose you enter abJk5 (valid id).
Then use arrow keys to go back 2 characters
Then click backspace and **immidiately** another character
The character gets inserted before some internal validation / transformation runs and therefore gets appended to the end of the string

Reply:
Hmm didn't noticed that but for some reason I had problems with this function before that it unfocused the input field when the id was valid which was weird as well and I could explain myself why

Reply:
I am very confused why the input field in general is at all unselected, can not figure out why this is the case at all